packages/docs-reanimated/docs/advanced/dispatchCommand.mdx

```diff
@@ -4,71 +4,82 @@ sidebar_position: 7
 
 # dispatchCommand
 
-import DocsCompatibilityInfo from '../_shared/_docs_compatibility_info.mdx';
+import DispatchCommandSrc from '!!raw-loader!@site/src/examples/DispatchCommand';
 
-<DocsCompatibilityInfo />
+`dispatchCommand` allows you to run commands on a native component. It happens on UI thread.
 
-Allows to dispatch command on a native component synchronously from the UI thread.
+## Reference
+
+```javascript
+import { dispatchCommand } from 'react-native-reanimated';
+
+function App() {
+  const animatedRef = useAnimatedRef();
+
+  const gesture = Gesture.Tap().onStart(() => {
+    // highlight-next-line
+    dispatchCommand(animatedRef, 'focus');
+  });
+
+  return (
+    <>
+      <AnimatedTextInput ref={animatedRef} style={styles.input} />
+      <GestureDetector gesture={gesture}>
+        <Button title="Focus" />
+      </GestureDetector>
+    </>
+  );
+}
+```
+
+<details>
+<summary>Type definitions</summary>
+
+```typescript
+function dispatchCommand<T extends Component>(
+  animatedRef: AnimatedRef<T>,
+  commandName: string,
+  args?: unknown[]
+) => void;
+```
+
+</details>
 
 ### Arguments
 
 #### `animatedRef`
 
-The product of [`useAnimatedRef`](/docs/core/useAnimatedRef) which is Reanimated's extension of a standard React ref (delivers the view tag on the UI thread).
+The product of [`useAnimatedRef`](/docs/core/useAnimatedRef), which is Reanimated's extension of a standard React ref (delivers the view tag on the UI thread).
 
-#### `commandName` [String]
+#### `commandName`
 
-The name of the command to execute, e.g. `focus` or `scrollToEnd`.
+The name of the command to execute, e.g. `'focus'` or `'scrollToEnd'`.
 
-#### `args` [Array<any\>]
+#### `args` <Optional />
 
 The array of command arguments. Defaults to an empty array.
 
-### Returns
+## Example
 
-`void`
+<ExampleVideo
+  sources={{
+    android: '/react-native-reanimated/recordings/dispatchCommand_android.mov',
+    ios: '/react-native-reanimated/recordings/dispatchCommand_ios.mov',
+  }}
+/>
```

Comment:
I wonder if we could attach here a more real use-case example instead of the current one.

You can simply touch the input boxes instead of buttons `focus login` and `focus pwd`, those buttons don't seem to provide any value.

Reply:
There is a value, someone might have some custom logic that focuses some certain text input - for example an error when filling some form automatically focuses said input that the user needs to fix.

Reply:
But yeah, even then the example would have to be changed a bit

Reply:
> There is a value, someone might have some custom logic that focuses some certain text input - for example an error when filling some form automatically focuses said input that the user needs to fix.

Then we should show exactly this use case. But don't worry, I gave @patrycjakalinska some suggestions already.